packages/material-ui/src/Unstable_TrapFocus/Unstable_TrapFocus.js

```diff
@@ -84,6 +84,15 @@ function Unstable_TrapFocus(props) {
         return;
       }
 
+      // A focused element that becomes unfocusable doesn't fire any events.
```

Comment:
In react specifically, no? By spec they should fire blur events.

Reply:
From what I remember of our exploration in #16585, we first try to listen to events inside, then outside React. We couldn't find any.

Reply:
I don't follow how these things relate.

Reply:
I think that the problem roots into:

```
// With Edge, Safari and Firefox, no focus related events are fired when the focused area stops being a focused area
 // e.g. https://bugzilla.mozilla.org/show_bug.cgi?id=559561.
//
// The whatwg spec defines how the browser should behave but does not explicitly mention any events:
// https://html.spec.whatwg.org/multipage/interaction.html#focus-fixup-rule.
```

In order to solve the missing events, we went with a interval check every 50ms. However, this interval logic is a bold move. I think that it can simplify the logic by limiting it to the bear minimum we need it for.

Note how this change allows to simplify the branch logic @mnajdova had to add.

Regarding #17321, after more considerations, I don't think that my fix handles all the cases, it's not ready yet. It seems that we would need to keep track of whether or not the focus was already moved into the trap. For instance with such a variable: https://github.com/theKashey/react-focus-lock/blob/26b2e928bbcb3fffaa0e843c07490c7e881b616e/src/Trap.js#L109.

What do you think of the following plan: I split the effort into two different pull requests. First, we limit the "boldness" of the interval call. Second, we solve the missing case do `disableAutoFocus`.

Reply:
So you agree that the comment is currently misleading? Can we change it so that it's apparent that this is a browser quirk that can't be normalized by React?

Reply:
Agree, I will update the comment